lib/elixir/lib/code.ex

```diff
@@ -424,6 +424,14 @@ defmodule Code do
     end
   end
 
+  defp module_attribute_or_atom(acc, rest) do
+    case Enum.reverse(rest) do
+      '' -> {:module_attribute, acc}
+      ':' ++ atom -> {:unquoted_atom, atom ++ '@' ++ acc}
+      _ -> :none
```

Comment:
What happens when you do: `foo :bar@baz` or `foo(:bar@baz)` or `foo :bar, @baz`? I am afraid in those cases simply looking at the rest won't be enough. I think what you probably need to do is to call `rest_identifier` again. If it returns `{_, _, '@' ++ _, _}` or `{_, _, ':' ++ _, _}`, then it is atom, otherwise it is a module attribute.

Reply:
Actually, you should call rest_identifier. If you get `{_, _, '@' ++ _, _}`, then call it again until you find a `{_, _, ':' ++ _, _}`. Otherwise it is a module attribute.

Reply:
you're correct about these cases, I'll try another iteration to fix them